Issue #12389: enforced fail-with-body in curl commands

rnveach opened the pull request:
Issue #12389

rnveach commented on .ci/close-create-milestone.sh:
```diff
@@ -7,10 +7,10 @@ source ./.ci/util.sh
 
 echo "Close previous milestone at github"
-MILESTONE_ID=$(curl -s \
+MILESTONE_ID=$(curl --fail-with-body -s \
```
Is it an issue to do this inside a `$()`?

romani replied:
Should not be issue, it is good to fail script if this curl failing

romani reviewed (approved):
Ok to merge

nrmancuso commented:
@rnveach looks like this option is not portable: https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=12007&view=logs&j=6a833393-c551-5000-0916-5000fd3b86e7&t=bdbbd129-77f0-5d73-ce7d-db58d34c8358

rnveach commented:
https://stackoverflow.com/questions/22506024/get-page-output-with-curl-fail
>  curl supports --fail-with-body option since v7.76.0

https://curl.se/changes.html#7_76_0
> 7.76.0 - March 31 2021 

I removed the command from this script. It is a good thing we split scripts on activity. Otherwise, I would have to use line numbers or it would end up suppressing the whole file.

rnveach commented:
https://app.travis-ci.com/github/checkstyle/checkstyle/jobs/588301476
Quarterly travis clean failed and there is nothing to say why. It also only happened on 1 node.

rnveach commented:
@romani I was checking if @nrmancuso was still ok with my change since I pushed a new change.
Travis failure is unrelated to this PR and I am examining some things in another PR.

If there is no complaints I can merge this.

nrmancuso reviewed (approved):
Ok to merge as CI passes
